Install the headers files into a subdirectory to avoid cluttering
the include directory with header files.

Signed-off-by: Rob Woolley <rob.woolley@windriver.com>

Index: git/gdal.cmake
===================================================================
--- git.orig/gdal.cmake
+++ git/gdal.cmake
@@ -830,7 +830,7 @@ install(
   ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
   LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
   RESOURCE DESTINATION ${GDAL_RESOURCE_PATH}
-  PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+  PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/gdal
   FRAMEWORK DESTINATION "${FRAMEWORK_DESTINATION}")
 
 if (NOT GDAL_ENABLE_MACOSX_FRAMEWORK)
